cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -3354,101 +3384,78 @@
           (x64_andnpd sign_bit a)
           (x64_andpd sign_bit b))))
 
-;; Rules for `ceil` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Helper for the `ceil`/`floor`/`nearest`/`trunc` instructions ;;;;;;;;;;;;;;;;
 
-(rule 1 (lower (ceil a @ (value_type $F32)))
+;; Emits either a `round{ss,sd,ps,pd}` instruction, as appropriate, or generates
+;; the appropriate libcall and sequence to call that.
+(decl x64_round (Type Value RoundImm) Xmm)
+(rule 1 (x64_round $F32 a imm)
         (if-let $true (use_sse41))
-        (x64_roundss a (RoundImm.RoundUp)))
-
-(rule 1 (lower (ceil a @ (value_type $F64)))
+        (x64_roundss a imm))
```

Comment:
I know we're going the other direction here but I started thinking, "what about the AVX versions of these?" Looking at `inst.isle`, they are emitted by `x64_round*`, so all is well.

Reply:
Yeah so far I feel like the current pattern of having a low-level instruction constructor which has an avx and a non-avx rule works out pretty well. It is relatively verbose and may be ripe for some future refactoring (e.g. auto-encoding as AVX if the context's feature flags are set) but for now it's nice to not have to worry about AVX at all really as part of `lower.isle`